core/src/main/java/hudson/model/FileParameterValue.java

```diff
@@ -151,30 +153,40 @@ public FileItem getFile() {
         return file;
     }
 
-    @Override
-    public BuildWrapper createBuildWrapper(AbstractBuild<?,?> build) {
-        return new BuildWrapper() {
-            @Override
-            public Environment setUp(AbstractBuild build, Launcher launcher, BuildListener listener) throws IOException, InterruptedException {
-            	if (!StringUtils.isEmpty(location) && !StringUtils.isEmpty(file.getName())) {
-            	    listener.getLogger().println("Copying file to "+location);
-                    FilePath ws = build.getWorkspace();
-                    if (ws == null) {
-                        throw new IllegalStateException("The workspace should be created when setUp method is called");
-                    }
-                    if (!ALLOW_FOLDER_TRAVERSAL_OUTSIDE_WORKSPACE && !ws.isDescendant(location)) {
-                        listener.error("Rejecting file path escaping base directory with relative path: " + location);
-                        // force the build to fail
-                        return null;
-                    }
-                    FilePath locationFilePath = ws.child(location);
-                    locationFilePath.getParent().mkdirs();
-            	    locationFilePath.copyFrom(file);
-                    locationFilePath.copyTo(new FilePath(getLocationUnderBuild(build)));
-            	}
-                return new Environment() {};
-            }
-        };
+    private void copyFileToWorkspace(Run<?,?> build) {
+        if (isFileCopied || StringUtils.isEmpty(location) || StringUtils.isEmpty(file.getName())) {
+            return;
+        }
+
+        TopLevelItem project = (TopLevelItem) build.project;
+        FilePath ws;
+        if (project instanceof FreeStyleProject && ((FreeStyleProject) project).getCustomWorkspace() != null) {
+            ws = new FilePath(new File(((FreeStyleProject) project).getCustomWorkspace()));
+        }
+        else {
+            ws = Jenkins.get().getWorkspaceFor(project);
```

Comment:
Nope, this is the workspace on the “master node”, which in a properly secured system is not used at all. There simply _is_ no equivalent of this concept for Pipeline builds—they can have zero or more workspaces on diverse nodes.